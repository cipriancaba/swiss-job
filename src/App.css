.App {
  display: flex;
}

.Sidebar {
  background-color: #111111;
  width: 250px;
  height: 100vh;
}

.Sidebar h3 {
  color: #999999;
  padding: 5px;
}

.Sidebar ul {
  list-style-type: none;
  padding: 0px;
}

.Sidebar li {
  line-height: 40px;
  text-indent: 20px;
}

.Sidebar li:hover {
  background-color: rgba(50, 50, 50, 0.6);
  color: white;
}

.Sidebar .selected {
  color: white;
  font-weight: bold;
}

.Sidebar a {
  display: block;
  text-decoration: none;
  color: #999999;
  font-size: larger;
}

.Content {
  flex: 8;
  padding: 10px;
}
