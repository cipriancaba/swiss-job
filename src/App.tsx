import * as React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import { PersistGate } from 'redux-persist/integration/react'
import './App.css'
import { appRoutes } from './appRoutes'
import { Sidebar } from './components/Sidebar'
import { persistor, store } from './configureStore'

class App extends React.Component {
  public render() {
    return (
      <Provider store={store}>
        <PersistGate loading={<div />} persistor={persistor}>
          <Router>
            <div className="App">
              <div className="Sidebar">
                <Sidebar routes={appRoutes} />
              </div>

              <div className="Content">
                {appRoutes.map(route => (
                  <Route key={route.path} exact={route.exact} path={route.path} component={route.content} />
                ))}
              </div>
            </div>
          </Router>
        </PersistGate>
      </Provider>
    )
  }
}

export default App
