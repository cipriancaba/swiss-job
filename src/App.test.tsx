import { configure, shallow } from 'enzyme'
import ReactSixteenAdapter from 'enzyme-adapter-react-16'
import * as React from 'react'
import * as ReactDOM from 'react-dom'
import App from './App'

configure({ adapter: new ReactSixteenAdapter() })

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<App />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders App', () => {
    const app = shallow(<App />)
    expect(app).toMatchSnapshot()
  })
})
