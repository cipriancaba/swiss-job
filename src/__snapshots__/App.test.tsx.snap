// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <PersistGate
        loading={<div />}
        persistor={
                Object {
                        "dispatch": [Function],
                        "flush": [Function],
                        "getState": [Function],
                        "pause": [Function],
                        "persist": [Function],
                        "purge": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
>
        <BrowserRouter>
                <div
                        className="App"
                >
                        <div
                                className="Sidebar"
                        >
                                <withRouter(SidebarComponent)
                                        routes={
                                                Array [
                                                        Object {
                                                          "content": [Function],
                                                          "exact": true,
                                                          "path": "/",
                                                          "sidebar": [Function],
                                                        },
                                                        Object {
                                                          "content": [Function],
                                                          "exact": true,
                                                          "path": "/manage",
                                                          "sidebar": [Function],
                                                        },
                                                        Object {
                                                          "content": [Function],
                                                          "exact": true,
                                                          "path": "/dataset",
                                                          "sidebar": [Function],
                                                        },
                                                      ]
                                        }
                                />
                        </div>
                        <div
                                className="Content"
                        >
                                <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/"
                                />
                                <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/manage"
                                />
                                <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/dataset"
                                />
                        </div>
                </div>
        </BrowserRouter>
</PersistGate>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <BrowserRouter>
          <div
                    className="App"
          >
                    <div
                              className="Sidebar"
                    >
                              <withRouter(SidebarComponent)
                                        routes={
                                                  Array [
                                                            Object {
                                                              "content": [Function],
                                                              "exact": true,
                                                              "path": "/",
                                                              "sidebar": [Function],
                                                            },
                                                            Object {
                                                              "content": [Function],
                                                              "exact": true,
                                                              "path": "/manage",
                                                              "sidebar": [Function],
                                                            },
                                                            Object {
                                                              "content": [Function],
                                                              "exact": true,
                                                              "path": "/dataset",
                                                              "sidebar": [Function],
                                                            },
                                                          ]
                                        }
                              />
                    </div>
                    <div
                              className="Content"
                    >
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/"
                              />
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/manage"
                              />
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/dataset"
                              />
                    </div>
          </div>
</BrowserRouter>,
        "loading": <div />,
        "persistor": Object {
          "dispatch": [Function],
          "flush": [Function],
          "getState": [Function],
          "pause": [Function],
          "persist": [Function],
          "purge": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="App"
>
            <div
                        className="Sidebar"
            >
                        <withRouter(SidebarComponent)
                                    routes={
                                                Array [
                                                            Object {
                                                              "content": [Function],
                                                              "exact": true,
                                                              "path": "/",
                                                              "sidebar": [Function],
                                                            },
                                                            Object {
                                                              "content": [Function],
                                                              "exact": true,
                                                              "path": "/manage",
                                                              "sidebar": [Function],
                                                            },
                                                            Object {
                                                              "content": [Function],
                                                              "exact": true,
                                                              "path": "/dataset",
                                                              "sidebar": [Function],
                                                            },
                                                          ]
                                    }
                        />
            </div>
            <div
                        className="Content"
            >
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/"
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/manage"
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/dataset"
                        />
            </div>
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Sidebar"
>
                <withRouter(SidebarComponent)
                                routes={
                                                Array [
                                                                Object {
                                                                  "content": [Function],
                                                                  "exact": true,
                                                                  "path": "/",
                                                                  "sidebar": [Function],
                                                                },
                                                                Object {
                                                                  "content": [Function],
                                                                  "exact": true,
                                                                  "path": "/manage",
                                                                  "sidebar": [Function],
                                                                },
                                                                Object {
                                                                  "content": [Function],
                                                                  "exact": true,
                                                                  "path": "/dataset",
                                                                  "sidebar": [Function],
                                                                },
                                                              ]
                                }
                />
</div>,
              <div
                className="Content"
>
                <Route
                                component={[Function]}
                                exact={true}
                                path="/"
                />
                <Route
                                component={[Function]}
                                exact={true}
                                path="/manage"
                />
                <Route
                                component={[Function]}
                                exact={true}
                                path="/dataset"
                />
</div>,
            ],
            "className": "App",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <withRouter(SidebarComponent)
                  routes={
                                    Array [
                                                      Object {
                                                        "content": [Function],
                                                        "exact": true,
                                                        "path": "/",
                                                        "sidebar": [Function],
                                                      },
                                                      Object {
                                                        "content": [Function],
                                                        "exact": true,
                                                        "path": "/manage",
                                                        "sidebar": [Function],
                                                      },
                                                      Object {
                                                        "content": [Function],
                                                        "exact": true,
                                                        "path": "/dataset",
                                                        "sidebar": [Function],
                                                      },
                                                    ]
                  }
/>,
                "className": "Sidebar",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "routes": Array [
                    Object {
                      "content": [Function],
                      "exact": true,
                      "path": "/",
                      "sidebar": [Function],
                    },
                    Object {
                      "content": [Function],
                      "exact": true,
                      "path": "/manage",
                      "sidebar": [Function],
                    },
                    Object {
                      "content": [Function],
                      "exact": true,
                      "path": "/dataset",
                      "sidebar": [Function],
                    },
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
/>,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/manage"
/>,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/dataset"
/>,
                ],
                "className": "Content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "/",
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "/manage",
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/manage",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "/dataset",
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/dataset",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <PersistGate
          loading={<div />}
          persistor={
                    Object {
                              "dispatch": [Function],
                              "flush": [Function],
                              "getState": [Function],
                              "pause": [Function],
                              "persist": [Function],
                              "purge": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
>
          <BrowserRouter>
                    <div
                              className="App"
                    >
                              <div
                                        className="Sidebar"
                              >
                                        <withRouter(SidebarComponent)
                                                  routes={
                                                            Array [
                                                                      Object {
                                                                        "content": [Function],
                                                                        "exact": true,
                                                                        "path": "/",
                                                                        "sidebar": [Function],
                                                                      },
                                                                      Object {
                                                                        "content": [Function],
                                                                        "exact": true,
                                                                        "path": "/manage",
                                                                        "sidebar": [Function],
                                                                      },
                                                                      Object {
                                                                        "content": [Function],
                                                                        "exact": true,
                                                                        "path": "/dataset",
                                                                        "sidebar": [Function],
                                                                      },
                                                                    ]
                                                  }
                                        />
                              </div>
                              <div
                                        className="Content"
                              >
                                        <Route
                                                  component={[Function]}
                                                  exact={true}
                                                  path="/"
                                        />
                                        <Route
                                                  component={[Function]}
                                                  exact={true}
                                                  path="/manage"
                                        />
                                        <Route
                                                  component={[Function]}
                                                  exact={true}
                                                  path="/dataset"
                                        />
                              </div>
                    </div>
          </BrowserRouter>
</PersistGate>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <BrowserRouter>
            <div
                        className="App"
            >
                        <div
                                    className="Sidebar"
                        >
                                    <withRouter(SidebarComponent)
                                                routes={
                                                            Array [
                                                                        Object {
                                                                          "content": [Function],
                                                                          "exact": true,
                                                                          "path": "/",
                                                                          "sidebar": [Function],
                                                                        },
                                                                        Object {
                                                                          "content": [Function],
                                                                          "exact": true,
                                                                          "path": "/manage",
                                                                          "sidebar": [Function],
                                                                        },
                                                                        Object {
                                                                          "content": [Function],
                                                                          "exact": true,
                                                                          "path": "/dataset",
                                                                          "sidebar": [Function],
                                                                        },
                                                                      ]
                                                }
                                    />
                        </div>
                        <div
                                    className="Content"
                        >
                                    <Route
                                                component={[Function]}
                                                exact={true}
                                                path="/"
                                    />
                                    <Route
                                                component={[Function]}
                                                exact={true}
                                                path="/manage"
                                    />
                                    <Route
                                                component={[Function]}
                                                exact={true}
                                                path="/dataset"
                                    />
                        </div>
            </div>
</BrowserRouter>,
          "loading": <div />,
          "persistor": Object {
            "dispatch": [Function],
            "flush": [Function],
            "getState": [Function],
            "pause": [Function],
            "persist": [Function],
            "purge": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="App"
>
              <div
                            className="Sidebar"
              >
                            <withRouter(SidebarComponent)
                                          routes={
                                                        Array [
                                                                      Object {
                                                                        "content": [Function],
                                                                        "exact": true,
                                                                        "path": "/",
                                                                        "sidebar": [Function],
                                                                      },
                                                                      Object {
                                                                        "content": [Function],
                                                                        "exact": true,
                                                                        "path": "/manage",
                                                                        "sidebar": [Function],
                                                                      },
                                                                      Object {
                                                                        "content": [Function],
                                                                        "exact": true,
                                                                        "path": "/dataset",
                                                                        "sidebar": [Function],
                                                                      },
                                                                    ]
                                          }
                            />
              </div>
              <div
                            className="Content"
              >
                            <Route
                                          component={[Function]}
                                          exact={true}
                                          path="/"
                            />
                            <Route
                                          component={[Function]}
                                          exact={true}
                                          path="/manage"
                            />
                            <Route
                                          component={[Function]}
                                          exact={true}
                                          path="/dataset"
                            />
              </div>
</div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="Sidebar"
>
                  <withRouter(SidebarComponent)
                                    routes={
                                                      Array [
                                                                        Object {
                                                                          "content": [Function],
                                                                          "exact": true,
                                                                          "path": "/",
                                                                          "sidebar": [Function],
                                                                        },
                                                                        Object {
                                                                          "content": [Function],
                                                                          "exact": true,
                                                                          "path": "/manage",
                                                                          "sidebar": [Function],
                                                                        },
                                                                        Object {
                                                                          "content": [Function],
                                                                          "exact": true,
                                                                          "path": "/dataset",
                                                                          "sidebar": [Function],
                                                                        },
                                                                      ]
                                    }
                  />
</div>,
                <div
                  className="Content"
>
                  <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/"
                  />
                  <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/manage"
                  />
                  <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/dataset"
                  />
</div>,
              ],
              "className": "App",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <withRouter(SidebarComponent)
                    routes={
                                        Array [
                                                            Object {
                                                              "content": [Function],
                                                              "exact": true,
                                                              "path": "/",
                                                              "sidebar": [Function],
                                                            },
                                                            Object {
                                                              "content": [Function],
                                                              "exact": true,
                                                              "path": "/manage",
                                                              "sidebar": [Function],
                                                            },
                                                            Object {
                                                              "content": [Function],
                                                              "exact": true,
                                                              "path": "/dataset",
                                                              "sidebar": [Function],
                                                            },
                                                          ]
                    }
/>,
                  "className": "Sidebar",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "routes": Array [
                      Object {
                        "content": [Function],
                        "exact": true,
                        "path": "/",
                        "sidebar": [Function],
                      },
                      Object {
                        "content": [Function],
                        "exact": true,
                        "path": "/manage",
                        "sidebar": [Function],
                      },
                      Object {
                        "content": [Function],
                        "exact": true,
                        "path": "/dataset",
                        "sidebar": [Function],
                      },
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/"
/>,
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/manage"
/>,
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/dataset"
/>,
                  ],
                  "className": "Content",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "/",
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "/manage",
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/manage",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "/dataset",
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/dataset",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
